STM32MP1xx add SPI LL

yboutreux opened the pull request:
CMSIS register modification are cherry-picked from Cube CMSIS v1.1.0.rc1, modifications non-related to SPI hadn't been selected.
Add SPI LL for STM32MP1x SoC.

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>

erwango committed:
ext: hal: st: stm32cube: Add HAL for the STM32H7x series

This commit adds the STM32CUBE HAL for STM32H7x series MCUs.

Origin: ST Microelectronics
License: BSD-3-Clause
URL: http://www.st.com/en/embedded-software/stm32cubeh7.html
Commit: 1.4.0
Purpose: HAL layer for STM32H7
Maintained-by: External

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango reviewed (changes requested):
Last commit miss update on stm32cube/stm32mp1xx/CMakeLists.txt

yboutreux commented:
> Last commit miss update on stm32cube/stm32mp1xx/CMakeLists.txt

Updated to add changes on stm32cube/stm32mp1xx/README

erwango committed:
stm32cube: L4: Re-enable CAN legacy driver

Following recent L4 cube package update, activation of CAN
legacy driver was disabled by mistake and new CAN driver was
enabled.
Since zephyr STM32 CAN driver is still using legacy API,
re-enable legacy API.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

FRASTM committed:
stm32cube: update stm32g0xx cube

One file being renamed:
driver/include/stm32g0xx_hal_conf_template.h
into:
driver/include/stm32g0xx_hal_conf.h

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on stm32cube/stm32mp1xx/README:
```diff
@@ -72,3 +72,10 @@ Patch List:
+    Impacted files:
+      ext/hal/st/stm32cube/stm32mp1xx/drivers/include/stm32mp1xx_ll_spi.h
+      ext/hal/st/stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c
+    ST Bug tracker ID: None
```
@yboutreux , can you give expected release version here?

yboutreux replied:
> @yboutreux , can you give expected release version here?

Modification done

arnopo committed:
stm32mp1: add i2c ll driver

Add I2C LL support on top of the V1.0 release.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>

yboutreux commented:
I also added the Cube version inside the commit

yboutreux committed:
stm32cube: stm32mp1xx: soc: Update SPI registers definition

Update CMSIS SPI registers definition

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>
Signed-off-by: Alexandre Torgue <alexandre.torgue@st.com>

yboutreux committed:
stm32cube: stm32mp1xx: drivers: add SPI's LL

Add SPI LL files for STM32MP1x.

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>
Signed-off-by: Alexandre Torgue <alexandre.torgue@st.com>

erwango committed:
stm32cube: Disable I2C HAL when defining I2C_SPEED_STANDARD

On some STM32 series, I2C HAL defines I2C_SPEED_STANDARD and
I2C_SPEED_FAST. These definitions conflict with Zephyr I2C API.

Since Zephyr I2C STM32 driver uses LL API, we can disable I2C HAL.
Disable I2C HAL for L1 and F2 series. Deactivation is already done
2 other impacted series F1 and F4.

Additionally, on F1 series, add the commented definition line to
make the change more visible.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
stm32cube: Disable Legacy CAN API

Remove fixes on Cube packages to use Legacy API.
Applied to STM32F0, F1, F3, F4, F7 and L4.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

atorgue commented on stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c:
```diff
@@ -0,0 +1,743 @@
+             (PrescalerParity << SPI_I2SCFGR_ODD_Pos));
+}
+
+/**
```
why those empty lines ?

yboutreux replied:
Picked directly code generated for CubeMP1

yboutreux commented:
PR closed since it went as an official PR on Zephyrproject
